morphia/src/test/java/org/mongodb/morphia/TestQuery.java

```diff
@@ -369,37 +391,67 @@ public void testElemMatchQueryCanBeNegated() {
 
         Mapper mapper = getMorphia().getMapper();
 
-        List<Key<PhotoWithKeywords>> keys = getDs().find(PhotoWithKeywords.class)
+        validate(pwk1, pwk2, pwk3, pwk4, mapper, getDs().find(PhotoWithKeywords.class)
```

Comment:
I was trying to avoid cutting too deeply but I bit the bullet and cleaned that up a bit.  less code now and easier to reason about what's being asserted.